### ISSUE TITLE
Site Editor: TT1-blocks doesn't support full width alignment

### ISSUE THREAD

**Issue by vindl** [Automattic/wp-calypso#49573]
FSE sites created with TT1-blocks theme don't show full width alignment blocks in editor properly (same goes for design step preview during signup).

<img width="1680" alt="Screenshot 2021-02-01 at 23 31 04" src="https://user-images.githubusercontent.com/1182160/106526374-c75d4900-64e5-11eb-86d9-05117da59bf4.png">

They seem to be rendered fine on front end however.

**Comment by mattwiebe** [comment]
It appears to be working but also now everything is ugly after #49479 landed

**Comment by mattwiebe** [comment]
Opened #49667 to track the general ugliness of tt1-blocks, appears to be related to running Gutenberg latest

**Comment by vindl** [comment]
@mattwiebe might be related to https://github.com/WordPress/gutenberg/issues/28581?

**Comment by mattwiebe** [comment]
Yup that's part of what I'm seeing. I thought the refresh of `tt1-blocks` might fix it, but only got part way there

**Comment by vindl** [comment]
After some of the latest updates to Gutenberg and related themes, this issue is still present.

**Comment by vindl** [comment]
cc @scruffian @jffng in case this needs to be fixed in tt1-blocks

**Comment by jeyip** [comment]
The issue seems to be with the `post-content` block. If you select `post-content` specifically and click on "Full width" alignment, everything else behaves as expected. For some reason, the post-content block in the tt1-blocks `page-home` template doesn't default to full width. We can address this problem a few ways:

1. We can update the tt1-block templates and add an `align: full` block attribute to the post content block. I drafted a PR [here](https://github.com/WordPress/theme-experiments/pull/205).
2. We can modify the `block.json` of the post-content gutenberg block to default to full width [here](https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/post-content/block.json)

The odd thing is that other instances of `post-content` blocks in block templates _do_ default to full width, but _don't_ require the `align: full` block attributes. This strange behavior could use more investigation.

All things considered, I'd opt for the first approach over the second. I'm reluctant to prescribe a default alignment for a block. I think that can wait for user testing and concrete use cases. The first approach would also be released faster, which would get this issue out of the way before the second round of site editor testing. We could get a quick fix into production, leave this issue open, and then circle back and investigate things more thoroughly.

I'm curious what others think

**Comment by Copons** [comment]
> The odd thing is that other instances of `post-content` blocks in block templates _do_ default to full width, but _don't_ require the `align: full` block attributes. This strange behavior could use more investigation.

@jeyip Do you have an example of where it works?
In my tests (all TT1 default templates containing Post Content), the block always starts at default width, which means that I can see this issue across all relevant templates. 🤔 

Either way, I think the current behaviour is working as expected: nested blocks can't be wider than parent blocks.
So if parent (e.g. Post Content) is on a default/narrow width, nested (e.g. Heading) will be at most as large as parent; setting it at full width won't do anything.

Considering the Post Content role, I'm actually inclined to default it to full width, freeing its content up to align as they see fit.
I might be overlooking something, but this is basically [the same we already do with the Group block](https://github.com/WordPress/theme-experiments/blob/5a02927f9198f903211aba44c009fb2ba793472d/tt1-blocks/block-templates/page-home.html#L3) wrapping the page content inside a `<main>` tag:
```html
<!-- wp:group {"tagName":"main", "align":"full"} -->
<main class="wp-block-group alignfull">
	<div class="wp-block-group__inner-container">
		<!-- wp:post-content /-->
	</div>
</main>
<!-- /wp:group -->
```

Although, I'm also reluctant to assign a default alignment to a block, as I think this is more of a theme responsibility.
We already `align: full` the Group block in all templates, I guess we should just do the same for Post Content 🤔

**Comment by jeyip** [comment]
> Do you have an example of where it works?
In my tests (all TT1 default templates containing Post Content), the block always starts at default width, which means that I can see this issue across all relevant templates. 🤔

Interesting! @Copons Let me double check 🤔

**Comment by jeyip** [comment]
@Copons I'm testing it locally and in production with fresh instances.

I checked the `single` template and the post content placeholder defaults to full width. 

<img width="1088" alt="Screen Shot 2021-02-12 at 9 00 04 AM" src="https://user-images.githubusercontent.com/5414230/107799042-c90ee400-6d11-11eb-9fe7-462d1d780580.png">

I am noticing, however, that the `page-home` post-content **placeholder** defaults to full width as well. 

<img width="1137" alt="Screen Shot 2021-02-12 at 9 06 18 AM" src="https://user-images.githubusercontent.com/5414230/107799054-cc09d480-6d11-11eb-988a-441974244849.png">

Maybe in the process of loading actual content, the post-content block alignment is being reset?

> Although, I'm also reluctant to assign a default alignment to a block, as I think this is more of a theme responsibility.
We already align: full the Group block in all templates, I guess we should just do the same for Post Content

If it turns out that I'm doing something wonky with my test environments and the other TT1 default templates have alignment issues, I agree with this approach 👍

**Comment by jeyip** [comment]
> Do you have an example of where it works?
In my tests (all TT1 default templates containing Post Content), the block always starts at default width, which means that I can see this issue across all relevant templates. 🤔

After more discussion and debugging with Jacopo, it turn out my testing environment was wonky! Post content blocks without the `align: full` attribute _do_ start at default widths, as expected.

> Considering the Post Content role, I'm actually inclined to default it to full width, freeing its content up to align as they see fit. I might be overlooking something, but this is basically the same we already do with the Group block wrapping the page content inside a < main > tag

This makes sense to me 👍

**Comment by jeyip** [comment]
I'll close out this issue after we regenerate the tt1-blocks previews in the FSE gutenboarding horizon flow

**Comment by jeyip** [comment]
Just merged a fix for the tt1-blocks design preview. The FSE gutenboarding horizon flow should display the correctly styled preview whenever the current mshot image cache expires.

According to @andrewserong, this happens in approximately 1 day. I'll double check production on Thursday to confirm, at which point I'll close out this issue if things look good.

**Comment by jeyip** [comment]
Looks like the TT1 design previews are loading correctly in production 🎉 . Closing out this issue, but feel free to reopen this if it's not the case.
